x/auth/client/rest/rest_test.go

```diff
@@ -324,21 +362,40 @@ func (s *IntegrationTestSuite) TestLegacyRestErrMessages() {
 
 	var txRes sdk.TxResponse
 	s.Require().NoError(val.ClientCtx.JSONMarshaler.UnmarshalJSON(out.Bytes(), &txRes))
+	s.Require().NotEqual(0, txRes.Code) // It fails, see comment above.
```

Comment:
i'd like to test a non-failing IBC tx (to reproduce #7281) , what's the easiest way to do it?

Reply:
@fedekunze @colin-axner ideas?

Reply:
I would recommend our testing package but I'm not sure it will play nice with inprocess integratin testing, it might though. Use [ConstructMsgCreateClient](https://github.com/cosmos/cosmos-sdk/blob/6344d626db1fbdba5e0f67425703c1584021bf5b/x/ibc/testing/chain.go#L412) to make an IBC message that will succeed and then submit it to your application

You'll need to, [create a coordinator and chain](https://github.com/cosmos/cosmos-sdk/blob/6344d626db1fbdba5e0f67425703c1584021bf5b/x/ibc/core/02-client/keeper/keeper_test.go#L73) first.

Otherwise if you need to do it via CLI, the solo machine create client would probably be the easiest. I tested it recently and there is a json pubkey arg in the help description you can use for the test.

Reply:
thanks @colin-axner! I went with the 2nd way, since the test was already using CLI.